### PR TITLE
Feature/fetchpriority

### PR DESCRIPTION
**Pull request by LPastine**
#### What problem is this solving?

The goal of this change is to improve LCP performance metric on Search Pages for desktop and mobile devices.

This is achieved by setting the Product Summary Images' **fetchpriority** to **high** for the first Product Summary on mobile and to the first three on desktop, and **fetchpriority** to **low** for the rest of the Product Summary Images. As well as setting **loading** to **eager** when this condition is fulfilled.

By following the [recommendation of CWV regarding the elimination of resource load delay](https://web.dev/articles/optimize-lcp#1_eliminate_resource_load_delay), it is expected to improve LCP metric on Search Pages on mobile devices, as on this context the first Product Summary Image is recurrently the LCP element, and this is expected to improve Load Delay metric.

To achieve this, it's needed that the Product Summary Context dispatches the Product Summary's position to effectively assess which is the first Product Summary Image. This was already approached on PR: https://github.com/vtex-apps/product-summary-context/pull/25

#### How to test it?

[Whirlpool Poland Workspace](https://www.whirlpool.pl/urzadzenia/pranie/pralki?workspace=fetchpriorityplp)

[Whirlpool France Workspace](https://www.whirlpool.fr/produits/lavage/lave-linge?workspace=fetchpriorityplp)

[Whirlpool Italy Workspace](https://www.whirlpool.it/prodotti/lavaggio-e-asciugatura/lavatrici?workspace=fetchpriorityplp)

[Hotpoint Italy Workspace](https://www.hotpoint.it/elettrodomestici/lavaggio-e-asciugatura/lavatrici?workspace=fetchpriorityplp)

[Bauknecht Germany Workspace](https://www.bauknecht.de/hausgeraete/waschen-trocknen/waschmaschinen?workspace=fetchpriorityplp)

[Hotpoint UK Workspace](https://www.hotpoint.co.uk/appliances/laundry/washing-machines?workspace=fetchpriorityplp)

#### Screenshots or example usage:
fetchpriority-plp

#### Related to / Depends on

Product Summary Context PR: https://github.com/vtex-apps/product-summary-context/pull/25

### CODE REVIEW

File: react/ProductSummaryImage.tsx [hunk 1]
```diff
@@ -440,6 +446,15 @@ function ProductImage({
               alt={name}
               className={imageClassname}
               onError={onError}
+              fetchpriority={
```
Comment (gvc) [inline react/ProductSummaryImage.tsx:449]: Should the case of no `position` be handled here?

**Review by gvc — CHANGES_REQUESTED** [review]
Could you document this and allow the fetchpriority to be set via Site Editor as well?

This PR is a good example. Thank you

File: react/ProductSummaryImage.tsx (hunk 1 above)
Comment (LPastine) [reply react/ProductSummaryImage.tsx:449]: Yes, I'll adapt the code for handling this case.

**Comment by LPastine** [comment]
> Could you document this and allow the fetchpriority to be set via Site Editor as well?
> 
> This PR is a good example. Thank you

Yes, I'll do it. I can't see any URL reference of the PR you're commenting but I have this PR as reference: https://github.com/vtex-apps/store-image/pull/85. I'll use this as reference and update with the changes.

**Comment by LPastine** [comment]
> Could you document this and allow the fetchpriority to be set via Site Editor as well?
> 
> This PR is a good example. Thank you

Added the documentation and the site editor configuration:

fetch

**Review by gvc — APPROVED** [review]

**Comment by gvc** [comment]
@LPastine last hurdle before merging, can you fix the linting errors:

```
/github/workspace/react/ProductSummaryImage.tsx
Error:   320:10  error    Replace `·product,·position·}:·{·product:·ProductSummaryTypes.Product,·position:·number·|·undefined` with `⏎····product,⏎····position,⏎··}:·{⏎····product:·ProductSummaryTypes.Product⏎····position:·number·|·undefined⏎·`                                        prettier/prettier
Error:   434:29  error    Replace `isMobile:·boolean,·position:·number·|·undefined` with `⏎····isMobile:·boolean,⏎····position:·number·|·undefined⏎··`                                                                                                                                      prettier/prettier
Error:   434:29  error    'isMobile' is already declared in the upper scope                                                                                                                                                                                                                 no-shadow
Error:   434:48  error    'position' is already declared in the upper scope                                                                                                                                                                                                                 no-shadow
Error:   436:22  error    Replace `·?·(position·===·1·?·'high'·:·'low')·:·position·<·4·?·'high'·:·'low';` with `⏎········?·position·===·1⏎··········?·'high'⏎··········:·'low'⏎········:·position·<·4⏎········?·'high'⏎········:·'low'`                                                     prettier/prettier
Warning:   438:5   warning  Expected blank line before this statement                                                                                                                                                                                                                         padding-line-between-statements
Error:   438:[17](https://github.com/vtex-apps/product-summary/actions/runs/7210587676/job/19677642065?pr=386#step:8:18)  error    Delete `;`                                                                                                                                                                                                                                                        prettier/prettier
Error:   439:4   error    Delete `;`                                                                                                                                                                                                                                                        prettier/prettier
Error:   464:30  error    Replace `fetchpriority·===·'byPosition'·?·getFetchPriority(isMobile,·position)·:·fetchpriority` with `⏎················fetchpriority·===·'byPosition'⏎··················?·getFetchPriority(isMobile,·position)⏎··················:·fetchpriority⏎··············`  prettier/prettier
Error:   573:68  error    Insert `,`                                                                                                                                                                                                                                                        prettier/prettier
Error:   576:29  error    Insert `,`                                                                                                                                                                                                                                                        prettier/prettier
Error:   578:28  error    Insert `,`                                                                                                                                                                                                                                                        prettier/prettier
```

**Comment by LPastine** [comment]
> @LPastine last hurdle before merging, can you fix the linting errors:
> 
> ```
> /github/workspace/react/ProductSummaryImage.tsx
> Error:   320:10  error    Replace `·product,·position·}:·{·product:·ProductSummaryTypes.Product,·position:·number·|·undefined` with `⏎····product,⏎····position,⏎··}:·{⏎····product:·ProductSummaryTypes.Product⏎····position:·number·|·undefined⏎·`                                        prettier/prettier
> Error:   434:29  error    Replace `isMobile:·boolean,·position:·number·|·undefined` with `⏎····isMobile:·boolean,⏎····position:·number·|·undefined⏎··`                                                                                                                                      prettier/prettier
> Error:   434:29  error    'isMobile' is already declared in the upper scope                                                                                                                                                                                                                 no-shadow
> Error:   434:48  error    'position' is already declared in the upper scope                                                                                                                                                                                                                 no-shadow
> Error:   436:22  error    Replace `·?·(position·===·1·?·'high'·:·'low')·:·position·<·4·?·'high'·:·'low';` with `⏎········?·position·===·1⏎··········?·'high'⏎··········:·'low'⏎········:·position·<·4⏎········?·'high'⏎········:·'low'`                                                     prettier/prettier
> Warning:   438:5   warning  Expected blank line before this statement                                                                                                                                                                                                                         padding-line-between-statements
> Error:   438:[17](https://github.com/vtex-apps/product-summary/actions/runs/7210587676/job/19677642065?pr=386#step:8:18)  error    Delete `;`                                                                                                                                                                                                                                                        prettier/prettier
> Error:   439:4   error    Delete `;`                                                                                                                                                                                                                                                        prettier/prettier
> Error:   464:30  error    Replace `fetchpriority·===·'byPosition'·?·getFetchPriority(isMobile,·position)·:·fetchpriority` with `⏎················fetchpriority·===·'byPosition'⏎··················?·getFetchPriority(isMobile,·position)⏎··················:·fetchpriority⏎··············`  prettier/prettier
> Error:   573:68  error    Insert `,`                                                                                                                                                                                                                                                        prettier/prettier
> Error:   576:29  error    Insert `,`                                                                                                                                                                                                                                                        prettier/prettier
> Error:   578:28  error    Insert `,`                                                                                                                                                                                                                                                        prettier/prettier
> ```

Yes sure, I'll fix them right away

**Comment by LPastine** [comment]
> @LPastine last hurdle before merging, can you fix the linting errors:
> 
> ```
> /github/workspace/react/ProductSummaryImage.tsx
> Error:   320:10  error    Replace `·product,·position·}:·{·product:·ProductSummaryTypes.Product,·position:·number·|·undefined` with `⏎····product,⏎····position,⏎··}:·{⏎····product:·ProductSummaryTypes.Product⏎····position:·number·|·undefined⏎·`                                        prettier/prettier
> Error:   434:29  error    Replace `isMobile:·boolean,·position:·number·|·undefined` with `⏎····isMobile:·boolean,⏎····position:·number·|·undefined⏎··`                                                                                                                                      prettier/prettier
> Error:   434:29  error    'isMobile' is already declared in the upper scope                                                                                                                                                                                                                 no-shadow
> Error:   434:48  error    'position' is already declared in the upper scope                                                                                                                                                                                                                 no-shadow
> Error:   436:22  error    Replace `·?·(position·===·1·?·'high'·:·'low')·:·position·<·4·?·'high'·:·'low';` with `⏎········?·position·===·1⏎··········?·'high'⏎··········:·'low'⏎········:·position·<·4⏎········?·'high'⏎········:·'low'`                                                     prettier/prettier
> Warning:   438:5   warning  Expected blank line before this statement                                                                                                                                                                                                                         padding-line-between-statements
> Error:   438:[17](https://github.com/vtex-apps/product-summary/actions/runs/7210587676/job/19677642065?pr=386#step:8:18)  error    Delete `;`                                                                                                                                                                                                                                                        prettier/prettier
> Error:   439:4   error    Delete `;`                                                                                                                                                                                                                                                        prettier/prettier
> Error:   464:30  error    Replace `fetchpriority·===·'byPosition'·?·getFetchPriority(isMobile,·position)·:·fetchpriority` with `⏎················fetchpriority·===·'byPosition'⏎··················?·getFetchPriority(isMobile,·position)⏎··················:·fetchpriority⏎··············`  prettier/prettier
> Error:   573:68  error    Insert `,`                                                                                                                                                                                                                                                        prettier/prettier
> Error:   576:29  error    Insert `,`                                                                                                                                                                                                                                                        prettier/prettier
> Error:   578:28  error    Insert `,`                                                                                                                                                                                                                                                        prettier/prettier
> ```

@gvc lint errors fixed!

**Comment by gvc** [comment]
@beatrizmaselli @LPastine there are conflicts, can you resolve them?

**Comment by beatrizmaselli** [comment]
> @beatrizmaselli @LPastine there are conflicts, can you resolve them?

solved :))

**Pull request merged**